### DISCUSSION TITLE
Update state change on slider end

### DISCUSSION THREAD

**Discussion by Jupihawx** [Kitware/trame#392] [Q&A] [answered]
Hi!

This might be a basic question, but I can't seem to figure out how to do so.
Currently, when a VSlider is interacted with and I set a @state.change(), it continuously callback throughout the interaction.
Is there a way to have the Slider use the callback function only when its released?

Thanks!

**Comment by jourdain** [comment, marked as answer]
For that you need to leverage [the events](https://v2.vuetifyjs.com/en/api/v-slider/#events) and you can either have the callback set the value on the JS side or bind your Python function.

```python
vuetify.VSlider(
   value=("var_name", 0),
   change="var_name = $event"
)
```

or 

```python
vuetify.VSlider(
   value=("var_name", 0),
   change=(on_slider_change_fn, "[$event]"),
)
```

**Reply by Jupihawx** [reply]
Thanks it works!
I am not so sure I understand why though, and I can't really find information. In the vuetify list you sent, how can I choose the event I'm interested in with your example? (And hence, why $event works as "end" in vuetify?)

**Reply by jourdain** [reply]
The list I provided has the type of `$event` when using `change, end, click...`.
$event in vue.js is a reserved name for what you get as argument when the method get called.

**Reply by Jupihawx** [reply]
Ah that is clear, thanks!
